connect-inject/endpoints_controller_test.go

```diff
@@ -804,7 +805,7 @@ func TestReconcileCreateEndpoint(t *testing.T) {
 			})
 			require.NoError(t, err)
 			defer consul.Stop()
-			consul.WaitForSerfCheck(t)
+			consul.WaitForServiceIntentions(t)
```

Comment:
This is brilliant, I totally messed this up when I put SerfCheck instead of ServiceIntentions, this is probably also the cause of the flakey tests I've seen today! Awesome work!